feat: Add pages for org auth tokens

mydea opened the pull request:
For now these pages do nothing, but it sets up the base UI behind feature flags where we can start filling in the actual functionality in follow up PRs.

For users with the `org-auth-tokens` feature, we rename the `Developer Settings` page under org settings to `Custom Integrations`, move it to a new section` Developer Settings` & add another subpage `Auth Tokens` there. 

![Screenshot 2023-06-02 at 10 22 43](https://github.com/getsentry/sentry/assets/2411343/6a3e1341-963b-405e-80f8-362826a12083)

~~We also add a direct link there in the account dropdown navigation:~~

I also referenced @priscilawebdev to maybe look over this from a react perspective, that I am not doing something stupid on that front xD feel free to re-assign this to another reviewer instead!

I haven't added any tests yet, will do this in the following steps when actually adding functionality.

ale-cota commented:
- I love the idea to have a Developer Settings Section in the general Settings page ⭐ adds way more clarity and makes it things more easily discoverable 
- I'm not so sure about link from the Account dropdown... whereas for user tokens it makes sense under the user
 settings, I don't think think this has the same "weight" as the other org level items. Plus a user can see more items in that dropdown, I'm afraid we're making it too crowded. I don't know how much people ever use those entry points to be honest.
 <img width="279" alt="Screenshot 2023-06-02 at 11 02 20" src="https://github.com/getsentry/sentry/assets/26696515/b8394367-444e-4ed9-9284-8a8e597efc4a">
- how about a link in the actual /settings page?
<img width="1278" alt="Screenshot 2023-06-02 at 10 59 19" src="https://github.com/getsentry/sentry/assets/26696515/6b848cff-255d-4e06-a58e-ee843820929d">

I'll be honest, I doubt the second or 3rd entry points will help a lot with discoverability, I think 1 is a great improvement and we need to make sure it's served/linked correctly in the related user flows.

mydea commented:
I can remove the link in the dropdown for now (we can always experiment with adding it later as well)! Also wasn't 100% sold, IMHO the main reason is to de-emphazise the user auth tokens - without this, you still may be more likely to stumble over the user tokens first and use them instead of the org ones? but not too sure about that...! 

For now, I'll remove them to streamline this - let's think about this some more!

mydea commented:
I also added it to the quick links overview:

![image](https://github.com/getsentry/sentry/assets/2411343/0b05d413-d204-4743-b7a3-8c5f2ebb74f6)

Note I also renamed it to `User Auth Tokens` and below from `Your Integrations` to `Custom Integrations` - I think that makes more sense here?

evanpurkhiser commented on static/app/views/settings/organization/navigationConfiguration.tsx:
```diff
@@ -100,4 +100,112 @@ const organizationNavigation: NavigationSection[] = [
   },
 ];
 
+export const organizationNavigationWithAuthTokens: NavigationSection[] = [
```
Did we just duplicate the entire nav tree here?

mydea replied:
Yes, I figured for this experimental phase this is fine/least likely to break? Would you prefer we mutate this in the feature flag case?

evanpurkhiser commented on static/app/views/settings/settingsIndex.spec.tsx:
```diff
@@ -55,7 +55,8 @@ describe('SettingsIndex', function () {
       slug: 'org-index',
-    } as Organization;
+      features: [],
+    } as unknown as Organization;
```
😬 why do you have to do this?

evanpurkhiser commented on static/app/views/settings/organizationAuthTokens/newAuthToken.tsx:
```diff
@@ -0,0 +1,41 @@
+  );
+}
+
+export default withOrganization(OrganizationAuthTokensNewAuthToken);
```
Are we not able to use `useOrganization` here?
